### PR TITLE
watch/events: Add eventInfo struct for brevity.

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by harshavardhana** [comment]
> why not merge Event and Source? It will be simpler.

Done!

File: cmd/client-s3.go [hunk 1]
```diff
@@ -448,76 +445,48 @@ func (c *s3Client) Watch(params watchParams) (*watchObject, *probe.Error) {
...
-							Port:      record.Source.Port,
-							UserAgent: record.Source.UserAgent,
-						},
+					eventChan <- EventInfo{
+						Time:      record.EventTime,
+						Size:      record.S3.Object.Size,
+						Path:      u.String(),
+						Client:    c,
+						Type:      EventCreate,
+						IP:        record.Source.Host,
```
Comment (krishnasrinivas) [inline cmd/client-s3.go]: `IP` can be renamed to `Host` to be consistent

File: cmd/watch-main.go [hunk 2]
```diff
@@ -97,8 +97,17 @@ func checkWatchSyntax(ctx *cli.Context) {
...
+		Time string    `json:"time"`
+		Size int64     `json:"size"`
+		Path string    `json:"path"`
+		Type EventType `json:"type"`
+	} `json:"events"`
+	Source struct {
+		IP        string `json:"ip,omitempty"`
+		Port      string `json:"port,omitempty"`
+		UserAgent string `json:"userAgent,omitempty"`
+	} `json:"source,omitempty"`
```
Comment (krishnasrinivas) [inline cmd/watch-main.go]: Status is not useful.

EvenInfo can be used for printing and `watchMessage` can be eliminated completely.

Comment (harshavardhana) [inline cmd/watch-main.go]: No we need the structure of json to be different. It is not a one to one mapping.

Comment (harshavardhana) [inline cmd/watch-main.go]: Status is consistent with all mc output it is part of json API.

File: cmd/client-s3.go (hunk 1 above)
Comment (harshavardhana) [inline cmd/client-s3.go]: Will do.

**Pull request merged**